Terraform deployment edits (#362)
google/exposure-notifications-server: docs/deploying.md

1. Run `terraform apply` to start deployment:

   Without Cloud Build Triggers:

   ```console
   terraform apply \
     -var project=$PROJECT_ID
   ```

   With Cloud Build Triggers:

   ```console
   terraform apply \
     -var project=${PROJECT_ID} \
     -var region="us-central-1" \
     -var use_build_triggers=true \
     -var repo_owner=${YOUR_REPO_OWNER} \
     -var repo_name=${YOUR_REPO_NAME}
   ```

Terraform will begin by creating the service accounts and enabling the services
on Google Cloud that are required to run this server.

#### Local development and testing example deployment

The default Terraform deployment is a production-ready, high traffic
deployment. For local development and testing, we recommend you use the
following sample deployment:

1. Run `terraform apply` with the following command:

   ```console
   terraform apply \
     -var project=${PROJECT_ID} \
     -var region="us-central-1" \
     -var use_build_triggers=true \
     -var repo_owner=${YOUR_REPO_OWNER} \
     -var repo_name=${YOUR_REPO_NAME} \
     -var cloudsql_tier="db-custom-1-3840" \
     -var cloudsql_disk_size_gb="16"
   ```

> 1. Initialize or migrate the database.

Reviewer: This is part of the broader steps. Is this getting grouped under the 'testing deployment'?

Author: Sorry misunderstood previous message, moved this into the main section, leaving the local dev and test section for only a scaled down deployment.

| exposure cleanup | cmd/cleanup-exposure | Deletes old exposure keys |
| export cleanup | cmd/cleanup-export | Deletes old exported files published by the exposure key export service |

### Deploying using Terraform

You can use Terraform to deploy the reference Exposure Notification on Google
Cloud. These instructions make use of the
[Google Cloud Terraform provider](https://github.com/terraform-providers/terraform-provider-google).

1. Download and install Terraform 0.12.

>    To install Terraform using Go, type:

Reviewer 2: You should _not_ do this. Terraform has pre-built binaries. Don't instruct users to compile from source.

Author: Removed as requested.

   repository name (for example, 'exposure-notification-server') and owner
   (for example 'google') as variables when running the `terraform apply`
   command.

1. Authenticate with Google Cloud using the `gcloud` command-line tool:

   ```console
   gcloud auth login && gcloud auth application-default login
   ```

   This will open two authentication windows in your web browser.

   You may need to `unset GOOGLE_APPLICATION_CREDENTIALS` as it takes precedence
   over gcloud's login settings.

1. Change to this directory and run `terraform init`.  Terraform will
   automatically download the plugins required to execute this code.

1. Run `terraform apply` to start deployment:

   Without Cloud Build Triggers:

   ```console
   terraform apply \
     -var project=$PROJECT_ID
   ```

   With Cloud Build Triggers:

   ```console
   terraform apply \
     -var project=${PROJECT_ID} \
     -var region="us-central-1" \
     -var use_build_triggers=true \
     -var repo_owner=${YOUR_REPO_OWNER} \
     -var repo_name=${YOUR_REPO_NAME}
   ```

Terraform will begin by creating the service accounts and enabling the services
on Google Cloud that are required to run this server.

> #### High traffic example deployment

Reviewer: ```suggestion
#### Scaling down the size of the deployment
```

> **Important**: This is a production-ready, high traffic sample deployment.

Reviewer 2: This is incorrect. The _default_ behavior is a a production-ready deployment. The example Terraform below is for a local development version. It's a smaller CloudSQL tier and database size.

Author: Thanks, edits applied.

> **Important**: This is a production-ready, high traffic sample deployment.

Reviewer: ```suggestion
**Important**: The default configuration will create a production-ready, high traffic deployment.
```

Author: Believe both of these edits have been incorporated in the latest branch commit. PTAL, cheers
